app/views/layouts/_navbar.html.erb

```diff
@@ -80,57 +83,58 @@
           <% end %>
         <% end %>
 
-        <% # Admin Dropdown %>
+        <%# Admin Dropdown %>
         <% if can? :read, :admin_panel %>
           <li class="divider-vertical"></li>
-            <li class="dropdown">
-                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Admin <b class="caret"></b>
-                </a>
-                <ul class="dropdown-menu">
-          <li><%= link_to "Announcements", announcements_path %></li>
-                  <li><%= link_to "Blackout Dates", blackouts_path %></li>
-                  <li><%= link_to "Reports", reports_path %></li>
-                  <li><%= link_to "Settings", edit_app_configs_path %></li>
-                </ul>
-              </li>
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">Admin <b class="caret"></b>
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to 'Announcements', announcements_path %></li>
+              <li><%= link_to 'Blackout Dates', blackouts_path %></li>
+              <li><%= link_to 'Reports', reports_path %></li>
+              <li><%= link_to 'Settings', edit_app_configs_path %></li>
+            </ul>
+          </li>
         <% end %>
 
-        <% # Rails Admin Link %>
+        <%# Rails Admin Link %>
         <% if can? :access, :rails_admin %>
-          <li class="divider-vertical"></li>
+          <li class="divider"></li>
           <li><%= link_to "Rails Admin", rails_admin_path %></li>
         <% end %>
-
-          <li class="divider-vertical"></li>
+        <li class="divider"></li>
       </ul>
 
-      <% # SEARCH BOX %>
-      <%= form_tag catalog_search_path, class: 'navbar-search pull-left', method: :get do %>
-          <%= search_field_tag 'query', (params[:query] unless params[:query].blank?), class: 'search-query span2', placeholder: "&#xf002;  ".html_safe %>
+      <%# SEARCH BOX %>
+      <%= form_tag catalog_search_path, class: 'navbar-form navbar-left', role: 'search', method: :get do %>
+        <div class="form-group">
+          <%= search_field_tag 'query', (params[:query] unless params[:query].blank?), class: 'form-control search-query', placeholder: "&#xf002; ".html_safe %>
+        </div>
       <% end %>
-      <% # Current User Dropdown %>
-      <ul class="nav pull-right" id="logged_in">
+      <%# Current User Dropdown %>
+      <ul class="nav navbar-nav navbar-right" id="logged_in">
+        <% if current_user %>
+          <li class="dropdown"><a href="#" class="dropdown-toggle" data-toggle="dropdown"><i class="fa fa-user"></i>
+            <%= current_user.name %>
+            <span class="caret"></b></a>
+            <ul class="dropdown-menu" role="menu">
+              <li><%= link_to "My Profile", current_user %></li>
+              <li><%= link_to "Log Out", destroy_user_session_path %></li>
+            </ul>
+          </li>
+        <% end %>
+      </ul>
       <% if current_user.nil? %>
```

Comment:
Right, but as I mentioned in my [other comment](https://github.com/YaleSTC/reservations/pull/1116/files#r24361008), the `ul` could be entirely within the first `if` clause (since it only applies when you're logged in) and then you can just use an `else` to display the "Sign In" `li`.